client/src/app/auth/firebase-auth.service.ts

```diff
@@ -33,35 +33,28 @@ export class FirebaseAuthService extends AuthService {
     return this.requireAuth().pipe(take(1));
   }
 
-  refreshToken() {
-    return fromPromise(<Promise<any>>firebase.auth().currentUser.getIdToken(true));
-  }
-
   signOut(): Observable<any> {
     return fromPromise(<Promise<any>>firebase.auth().signOut());
   }
 
-  signInWithGitHub(): Observable<LoginUser> {
+  signInWithCredential(credential: firebase.auth.AuthCredential): Observable<LoginUser> {
     const loginPromise = firebase
       .auth()
-      .signInWithPopup(new firebase.auth.GithubAuthProvider())
+      .signInAndRetrieveDataWithCredential(credential)
```

Comment:
Had no idea this API existed! That's the thing that lets us sign in without producing a Popup - nice!